5804: Prints file name of invalid json file when parsing

renato2099 opened the pull request:
## Description
Enhances handling runtime exception thrown when parsing JSON by adding file name to the exception message.
This was proposed in https://github.com/apache/incubator-pinot/issues/5804

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
No

Does this PR fix a zero-downtime upgrade introduced earlier?
No

Does this PR otherwise need attention when creating release notes? 
No

Jackie-Jiang commented:
It is better to log the bad file name in the `CreateSegmentCommand.java` under the catch of line 251, where bad files of all types can be logged

renato2099 commented:
I was not sure which were the code paths that could use the json parsing, that is why I added it where it actually takes place.
I moved it now to the createSegmentCommand as per your suggestion @Jackie-Jiang  I decided to keep the test as I think it is good that we add tests for these commands

Jackie-Jiang commented on pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/CreateSegmentCommand.java:
```diff
@@ -252,7 +252,8 @@ public boolean execute()
               LOGGER.warn("Caught exception while creating/verifying segment, will retry", e);
             } else {
-              throw e;
+              final String exceptionMessage = String.format("%s. File: %s", e.getMessage(), dataFiles.get(sequenceId));
+              throw new Exception(exceptionMessage, e);
```
```suggestion
              throw new RuntimeException("Caught exception while generating segment from file: " + dataFiles.get(sequenceId), e);
```

renato2099 replied:
updated

Jackie-Jiang commented on pinot-tools/src/test/java/org/apache/pinot/tools/TestCreateSegmentCommand.java:
```diff
@@ -0,0 +1,64 @@
+  )
+  public void testReadingInvalidJsonFile()
+      throws Exception {
+    final String fileDirectoryPath = JSON_INVALID_SAMPLE_DATA_FILE.getParent();
```
(nit) We don't usually put `final` for local variables

renato2099 replied:
sure, btw is there a code style guide where I can check these things?

Jackie-Jiang replied:
We use the style config file located at `config/codestyle-intellij.xml` (`config/codestyle-eclipse.xml` for eclipse)
Will add some code style guide in the Pinot doc
